src/web/error.rs

```diff
@@ -32,90 +38,173 @@ pub enum AxumNope {
     Redirect(String, CachePolicy),
 }
 
-impl IntoResponse for AxumNope {
-    fn into_response(self) -> AxumResponse {
+// FUTURE: Ideally, the split between the 3 kinds of responses would
+// be done by having 3 kinds of enums in the first place instead of
+// just `AxumNope`, to keep the line statically type-checked
+// throughout instead of having the conversion?
+
+impl AxumNope {
```

Comment:
I slept over it and would like to change the design a little bit. 

I'll try to summarize it, but feel free to ask any follow-up questions. 

- `AxumNope` gets a private method `error_info` which just returns the `ErrorInfo` for the cases we have some. `NoResults` and `Redirect` can use `unreachable!`. 
- the `IntoResponse` impl for `AxumNope` matches on `NoResults` & `Redirect` first, and defaults to generating a response using `error_info` 
- the `IntoResponse` impl for `AxumJsonNope` does the same, and we should also generate API responses for the `Redirect` & `NoResults` case (IMO a normal HTTP redirect in the former case, and a `404` in the latter)

Reply:
Done. It saved 11 lines of code, 7 comment lines, and the indirection
via intermediate type, but adds a function (`redirect_with_policy`)
instead. It's not what I thought of first, but I think it's a win.